manta-pay/concrete-protocol.tex

```diff
@@ -4,46 +4,85 @@
+
+To use group elements in affine form we also define the projections:
+\[\X : \mathbb{G} \to \mathbb{F} \,\,\,\text{and}\,\,\, \Y : \mathbb{G} \to \mathbb{F}\]
+which we use below to insert group elements into field-only hash functions.
+
+For this protocol, we use \texttt{BLS12-381} as our outer (pairing-friendly) curve with scalar field $\mathbb{F}$ and \texttt{JubJub} as our inner curve with scalar field $\mathbb{S}$.
 
 \subsection{Concrete Cryptographic Schemes}
 
-\begin{definitiontoc}{Commitment Schemes}
-    The protocol features two different commitment schemes: $\COM^{\UTXO}$ the \UTXO{} Commitment Scheme and $\COM^{\VN}$ the Void Number Commitment Scheme. Both commitment schemes use \Poseidon{} as the cryptographic primitive. The \UTXO{} uses an arity-4 \Poseidon{} with the following mapping:
-    \[\COM^{\UTXO}(\esk, \pk, \asset) := \Poseidon_4(\lift(\esk), x(\pk), \asset.\ID, \asset.\VALUE)\]
-    where $x$ is the coordinate-extraction function from $\mathbb{G}$ to $\mathbb{F}$.
-    For the Void Number Commitment Scheme we use an arity-2 \Poseidon{} with the following mapping:
-    \[\COM^{\VN}(\sk, \cm) := \Poseidon_2(\lift(\sk), \cm)\]
-\end{definitiontoc}
+\begin{definition}[Commitment Schemes]
+    The protocol features two different commitment schemes: $\COM^{\UTXO}$ the \UTXO{} Commitment Scheme and $\COM^{\VN}$ the Void Number Commitment Scheme. Both commitment schemes use \Poseidon{} as the underlying cryptographic primitive. The \UTXO{} uses an arity-8 \Poseidon{} with the following mapping:
+    \[\COM^{\UTXO}_r(\textsf{D}, \pkD, \asset) := \Poseidon_8(d, 0, r, \X(\textsf{D}), \Y(\textsf{D}), \X(\pkD), \Y(\pkD), \asset.\ID, \asset.\VALUE)\]
+    where $d = \HashToScalar(\domainstr{manta-pay/1.0.0/com-utxo})$.
+    For the Void Number Commitment Scheme we use an arity-4 \Poseidon{} with the following mapping:
+    \[\COM^{\VN}_\ak(\cm) := \Poseidon_4(\HashToScalar(\domainstr{manta-pay/1.0.0/com-vn}), 0, \X(\ak), \Y(\ak), \cm)\]
+\end{definition}
+
+\begin{definition}[Key-Derivation Functions]
+    For the encryption scheme \KDF{s}, we use the following which maps a group element $G : \mathbb{G}$ to a scalar:
+    \[\KDF(G) := \Poseidon_2(\HashToScalar(\domainstr{manta-pay/1.0.0/encryption-kdf}), x(G), y(G))\]
+\end{definition}
+
+\begin{definition}[Randomizable Key-Derivation Function]
+    For \rKDF, we use the following which uses a scalar $r: \mathbb{S}$ to randomize a scalar $x: \mathbb{S}$ to a scalar 
+    and a group element $G : \mathbb{G}$ to a group element:
+    \begin{align*}
+        \rKDF.\rand^I(r, x): \mathbb{S} \times \mathbb{S} \to \mathbb{S} := r * x \\
+        \rKDF.\rand^O(r, G): \mathbb{S} \times \mathbb{G} \to \mathbb{G} := r \cdot G
+    \end{align*}
+
```

Comment:
Maybe add a remark about possibly ill-chosen r values, e.g. divisors of the order of G.

Reply:
Need to describe sampling methods for randomness/blinder-type arguments (usually uniformly sampled).